sim/rogue/item_sets_phase_8.go

```diff
@@ -7,16 +7,104 @@ import (
-	aura.ApplyOnInit(func(aura *core.Aura, sim *core.Simulation) {
-		rogue.OnComboPointsSpent(func(sim *core.Simulation, spell *core.Spell, comboPoints int32) {
-			if spell.ProcMask != core.ProcMaskEmpty && sim.Proc(0.20*float64(comboPoints), "Fallen Regality Proc") {
-				rogue.AddEnergy(sim, 20, energyMetrics)
-			}
-		})
+var ItemSetDuskwraithArmor = core.NewItemSet(core.ItemSet{
+	Name: "Duskwraith Armor",
+	Bonuses: map[int32]core.ApplyEffect{
+		2: func(agent core.Agent) {
+			rogue := agent.(RogueAgent).GetRogue()
+			rogue.applyScarletEnclaveDamage2PBonus()
+		},
+		4: func(agent core.Agent) {
+			rogue := agent.(RogueAgent).GetRogue()
+			rogue.applyScarletEnclaveDamage4PBonus()
+		},
+		6: func(agent core.Agent) {
+			rogue := agent.(RogueAgent).GetRogue()
+			rogue.applyScarletEnclaveDamage6PBonus()
+		},
+	},
+})
+
+// 2P
+// While Just a Flesh Wound is not active, your Backstab, Sinister Strike, Saber Slash, and Mutilate deal 10% increased damage per your active Poison or Bleed effect
+// afflicting the target, up to a maximum increase of 30%
+func (rogue *Rogue) applyScarletEnclaveDamage2PBonus() {
+	if rogue.HasRune(proto.RogueRune_RuneJustAFleshWound) {
+		return
+	}
+
+	label := "S03 - Item - Scarlet Enclave - Rogue - Damage 2P Bonus"
+	if rogue.HasAura(label) {
+		return
+	}
+
+	spellsModifiedBySetBonus := ClassSpellMask_RogueBackstab | ClassSpellMask_RogueSinisterStrike | ClassSpellMask_RogueSaberSlash | ClassSpellMask_RogueMutilateHit
+
+	damageMod := rogue.AddDynamicMod(core.SpellModConfig{
+		Kind:       core.SpellMod_DamageDone_Pct,
+		ClassMask:  spellsModifiedBySetBonus,
+		FloatValue: 1.0,
 	})
+
+	core.MakeProcTriggerAura(&rogue.Unit, core.ProcTrigger{
+		ActionID:       core.ActionID{SpellID: 1226843},
+		Name:           label,
+		Callback:       core.CallbackOnApplyEffects,
+		ClassSpellMask: spellsModifiedBySetBonus,
+		Handler: func(sim *core.Simulation, spell *core.Spell, result *core.SpellResult) {
+			totalBleedsAndPoisons := rogue.PoisonsActive[rogue.CurrentTarget.UnitIndex] + rogue.BleedsActive[rogue.CurrentTarget.UnitIndex]
+
+			// Only apply the damage mod up to 3 times for the 30% bonus maximum
+			damageMod.UpdateFloatValue(1 + 0.10*float64(min(3, totalBleedsAndPoisons)))
+			damageMod.Activate()
+		},
+	})
+}
+
+// 4P
+// Your Poison and autoattack critical strikes have a 10% chance to grant you a combo point. (Proc chance: 10%)
+func (rogue *Rogue) applyScarletEnclaveDamage4PBonus() {
+	label := "S03 - Item - Scarlet Enclave - Rogue - Damage 4P Bonus"
+	if rogue.HasAura(label) {
+		return
+	}
+
+	// Combo! spell that adds the combo point for the set bonus
+	comboPointMetrics := rogue.NewComboPointMetrics(core.ActionID{SpellID: 1226869})
+
+	procClassMasks := ClassSpellMask_RogueDeadlyPoisonTick | ClassSpellMask_RogueOccultPoisonTick | ClassSpellMask_RogueInstantPoison
+
+	core.MakePermanent(rogue.RegisterAura(core.Aura{
+		Label: label,
```

Comment:
Sorry I thought you were adding the ActionID here or I would've mentioned it sooner! To expose this as the 6pc aura we'd want to add
```
ActionID: core.ActionID{SpellID: 1226869}
```